test/gtest/fused_conv_bias_res_add_activ.cpp

```diff
@@ -25,14 +25,44 @@
  *******************************************************************************/
 #include <gtest/gtest.h>
 #include <miopen/miopen.h>
+#include <miopen/env.hpp>
+#if MIOPEN_USE_COMPOSABLEKERNEL
+#include <miopen/solver/ck_utility_common.hpp>
+#endif
 
 #include "tensor_util.hpp"
 #include "get_handle.hpp"
 
 #include "conv3d_test_case.hpp"
 
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+
+#if MIOPEN_USE_COMPOSABLEKERNEL
+#define WORAROUND_ISSUE_2533 1
+#endif
+
 namespace conv_bias_act_res_add_fwd {
 
+bool TestIsApplicable()
+{
+#if MIOPEN_USE_COMPOSABLEKERNEL
+    const auto float_arg = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
+    return
+#if WORAROUND_ISSUE_2533
+        miopen::solver::ck_utility::is_ck_whitelist(get_handle().GetDeviceName()) //
+#else
+    /// \todo Check against specific ASCIs.
+#endif
+        && (float_arg == "--half" // So far only test for fp16 is implemented.
+            || float_arg.empty()) // Empty when gtest is run without parameters.
+        && !miopen::IsDisabled(
+               ENV(MIOPEN_TEST_ALL)); // Not disabled when gtest is run without parameters.
```

Comment:
I would like to remove this dependence on environment variables like `MIOPEN_TEST_ALL` or `MIOPEN_TEST_FLOAT_ARG` for new tests. All new tests should have unit tests methods for all the supported types. 

CC: @JehandadKhan , @junliume

Reply:
This is for CI stage 'Fp32 Hip Debug NOCK AnyGPU Build-Only' to pass.

Reply:
@amberhassaan I'm second to this. We will remove this when possible. However, for now we need to handle `MIOPEN_TEST_FLOAT_ARG` to support the existing CI infrastructure, which distributes testing of different datatypes on different nodes (for the sake of faster CI turnaround). See #1843.

Ditto `MIOPEN_TEST_ALL` because we still want to have smoke tests in our CI.